### PR TITLE
Fix headers on publishing error response

### PR DESCRIPTION
**Pull request by velitchko-valkov**
[#32] Fix headers on publishing error response
- Added error response code
- Optimized error response headers

Signed-off-by: Velitchko Valkov <Velitchko.Valkov@bosch.io>

### CODE REVIEW

File: client/uploadable.go [hunk 1]
```diff
@@ -296,14 +306,17 @@ func (u *AutoUploadable) messageHandler(requestID string, msg *protocol.Envelope
...
 		logger.Errorf("error while executing operation %s: %s", operation, responseError.Message)
 	}
 
+	headers := protocol.NewHeaders(protocol.WithCorrelationID(msg.Headers.CorrelationID()),
+		protocol.WithContentType("application/json"))
+
 	reply := &protocol.Envelope{
 		Topic:   msg.Topic,                                           // preserve the topic
-		Headers: msg.Headers,                                         // preserve the headers
+		Headers: headers,                                             // preserve the headers
```
Comment (e-grigorov) [inline client/uploadable.go:319]: The headers are not preserved any more. The comment is not correct.

File: client/uploadable.go [hunk 2]
```diff
@@ -95,10 +95,24 @@ type AutoUploadable struct {
...
+	ErrorCodeParameterInvalid ErrorCode = "messages:parameter.invalid"
+	ErrorCodeExecutionFailed            = "messages:execution.failed"
+)
+
 //ErrorResponse is returned from operations handling functions
 type ErrorResponse struct {
-	Status  int
-	Message string
+	Status  int       `json:"status"`
+	Code    ErrorCode `json:"error"`
```
Comment (e-grigorov) [inline client/uploadable.go:110]: Actually, the struct field "Code" is not correct. It should be: ErrorCode or Error for short but that's not a generic code.

**Review by e-grigorov — CHANGES_REQUESTED** [review]

File: client/uploadable.go (hunk 1 above)
Comment (velitchko-valkov) [reply client/uploadable.go]: Okay, I will adjust the comment to suit the new case.

File: client/uploadable.go (hunk 2 above)
Comment (velitchko-valkov) [reply client/uploadable.go]: I will change the field name to "ErrorCode".

**Review by e-grigorov — APPROVED** [review]

**Pull request merged**